src/preprocessing/preprocessing-modules/BrowserDetection.ts

```diff
@@ -0,0 +1,45 @@
+import * as Bowser from 'bowser';
+import { Config } from '../../config';
+
+/**
+ * Based on:
+ * - https://caniuse.com/#feat=mdn-http_headers_csp_content-security-policy
+ * - https://caniuse.com/#feat=mdn-http_headers_csp_content-security-policy_worker-src (most recent CSP feature used)
+ * - https://caniuse.com/#feat=rel-noopener
+ * - https://caniuse.com/#feat=rel-noreferrer
+ * - https://caniuse.com/#feat=iframe-sandbox
+ * - https://caniuse.com/#feat=mdn-html_elements_iframe_sandbox-allow-popups-to-escape-sandbox (most recent sandbox feature used)
+ * - https://caniuse.com/#feat=mdn-api_htmlframeelement_contentwindow
+ * - https://caniuse.com/#feat=url
+ * - https://caniuse.com/#feat=getrandomvalues
+ */
+const BROWSER_REQUIREMENTS = {
```

Comment:
Is this mobile or desktop? would it work with Edge?

Reply:
Good point! I decided to add much more comprehensive tests for this.

In short, this covers both desktop and mobile by default. I added an exception for iOS Safari as it doesn't support `rel=noopener` before 10.3 it seems.

Edge is not supported, even the new version, as it still doesn't seem to support all the CSP values. I'll investigate this in the future (#35).